### PR TITLE
Add typedocs for the Token module

### PR DESCRIPTION
**Pull request by Tschakki**
### What was the problem?

This PR resolves https://github.com/LiskHQ/lisk-docs/issues/1543

### How was it tested?

`npx typedoc`

### CODE REVIEW

File: framework/src/modules/base_command.ts [hunk 1]
```diff
@@ -28,7 +31,28 @@ export abstract class BaseCommand<T = unknown> {
 	// eslint-disable-next-line no-useless-constructor
 	public constructor(protected stores: NamedRegistry, protected events: NamedRegistry) {}
 
+	/**
+	 * The hook `Command.verify()` is called to do all necessary verifications.
+	 * If the verification of the command was successful, the command can be {@link execute | executed} as next step.
```
Comment (chris529) [inline framework/src/modules/base_command.ts:36]:
```suggestion
	 * If the verification of the command was successful, for the next step the command can be {@link execute | executed}.
```

File: framework/src/modules/base_command.ts [hunk 2]
```diff
@@ -28,7 +31,28 @@ export abstract class BaseCommand<T = unknown> {
...
+	 * Similar to the {@link BaseModule.verifyTransaction} hook, `Command.verify()` will be called also in the {@link @liskhq/lisk-transaction-pool!TransactionPool}, and it is to ensure the verification defined in this hook is respected when the transactions are included in a block.
```
Comment (chris529) [inline framework/src/modules/base_command.ts:37]:
```suggestion
	 * Similar to the {@link BaseModule.verifyTransaction} hook, `Command.verify()` will be called also in the {@link @liskhq/lisk-transaction-pool!TransactionPool}, and Its purpose is to guarantee that the verification defined within this hook is adhered to when the transactions are incorporated into a block.
```

File: framework/src/modules/base_command.ts [hunk 3]
```diff
@@ -28,7 +31,28 @@ export abstract class BaseCommand<T = unknown> {
...
 
+	/**
+	 * Applies the state changes of a command through the state machine.
+	 * The hook `Command.execute()` is triggered by a transaction identified by the module name and the command name.
+	 *
+	 * Additionally, an event will be emitted that provides the information on whether a command is executed successfully or failed.
+	 *
+	 * In this hook, the *state can be mutated* and *events* can be emitted.
+	 *
+	 * If the {@link verify | command verification} succeeded, but the hook execution *fails*, the transaction that triggered this command is still valid, but the *state changes applied in this hook are reverted.*
```
Comment (chris529) [inline framework/src/modules/base_command.ts:53]:
```suggestion
	 * If the {@link verify | command verification} succeeded, but the hook execution *fails*, the transaction that triggered this command is still valid, however the *state changes applied in this hook are reverted.*
```

File: framework/src/modules/interoperability/base_cc_command.ts [hunk 4]
```diff
@@ -16,16 +16,42 @@ import { Schema, emptySchema } from '@liskhq/lisk-codec';
...
+	 */
 	public get name(): string {
 		const name = this.constructor.name.replace('CCCommand', '');
 		return name.charAt(0).toLowerCase() + name.substring(1);
 	}
 
 	// eslint-disable-next-line no-useless-constructor
 	public constructor(protected stores: NamedRegistry, protected events: NamedRegistry) {}
+	/**
+	 * The hook `CCCommand.verify()` is called to do all necessary verifications.
```
Comment (chris529) [inline framework/src/modules/interoperability/base_cc_command.ts:36]:
```suggestion
	 * The hook `CCCommand.verify()` is called to perform all necessary verifications.
```

File: framework/src/modules/interoperability/base_cc_command.ts [hunk 5]
```diff
@@ -16,16 +16,42 @@ import { Schema, emptySchema } from '@liskhq/lisk-codec';
...
+	 *
+	 * In this hook, the state *cannot* be mutated and events cannot be emitted.
+	 *
+	 * If the verification of the command was successful, the cc-command can be {@link execute | executed} as next step.
```
Comment (chris529) [inline framework/src/modules/interoperability/base_cc_command.ts:40]:
```suggestion
	 * If the verification of the command was successful, for the next step the cc-command can be {@link execute | executed}.
```

File: framework/src/modules/interoperability/constants.ts [hunk 6]
```diff
@@ -43,6 +43,13 @@ export const MIN_MODULE_NAME_LENGTH = 1;
 export const MAX_MODULE_NAME_LENGTH = 32;
 export const MIN_CROSS_CHAIN_COMMAND_NAME_LENGTH = 1;
 export const MAX_CROSS_CHAIN_COMMAND_NAME_LENGTH = 32;
+/**
+ * Chain identifiers are 4-byte values that follow a specific format:
+ * - The network-specific prefix: The first byte is used to identify the mainchain and its corresponding network of blockchains to which the chain wishes to connect.
```
Comment (chris529) [inline framework/src/modules/interoperability/constants.ts:48]:
```suggestion
 * - The network-specific prefix: The first byte is used to identify the mainchain and its corresponding network of blockchains to which the chain wishes to establish a connection.
```

File: framework/src/modules/token/cc_commands/cc_transfer.ts [hunk 7]
```diff
@@ -25,21 +25,48 @@ import { EscrowStore } from '../stores/escrow';
...
+	 */
 	public init(args: { tokenMethod: TokenMethod; internalMethod: InternalMethod }): void {
 		this._tokenMethod = args.tokenMethod;
 		this._internalMethod = args.internalMethod;
 	}
 
+	/**
+	 * Checks if the token is native on either the sending or the receiving chain.
+	 *
+	 * If the token is native to the receiving chain, it also checks if there is enough balance in the escrow account to transfer the specified tokens.
```
Comment (chris529) [inline framework/src/modules/token/cc_commands/cc_transfer.ts:64]:
```suggestion
	 * If the token is native to the receiving chain, it additionally verifies whether the escrow account holds a sufficient balance to facilitate the transfer of the designated tokens.
```

File: framework/src/modules/token/commands/transfer.ts [hunk 8]
```diff
@@ -27,23 +27,50 @@ import { TokenID } from '../types';
...
+ */
 export class TransferCommand extends BaseCommand {
 	public schema = transferParamsSchema;
 	private _method!: TokenMethod;
 	private _internalMethod!: InternalMethod;
 
+	/**
+	 * The `init()` hook of a command is called by the Lisk Framework when the node starts.
+	 *
+	 * Here, you can validate and cache the module config or do initializations which should only happen once per node starts.
```
Comment (chris529) [inline framework/src/modules/token/commands/transfer.ts:56]:
```suggestion
	 * In this context, you have the opportunity to validate and cache the module config or perform initializations that are intended to occur only once when each node starts. 
```

File: framework/src/modules/token/commands/transfer_cross_chain.ts [hunk 9]
```diff
@@ -47,13 +47,28 @@ interface Params {
...
 	public schema = crossChainTransferParamsSchema;
 	private _moduleName!: string;
 	private _method!: TokenMethod;
 	private _interoperabilityMethod!: InteroperabilityMethod;
 	private _internalMethod!: InternalMethod;
 
+	/**
+	 * The `init()` hook of a command is called by the Lisk Framework when the node starts.
+	 *
+	 * Here, you can validate and cache the module config or do initializations which should only happen once per node starts.
```
Comment (chris529) [inline framework/src/modules/token/commands/transfer_cross_chain.ts:66]:
```suggestion
	 * In this context, you have the opportunity to validate and cache the module config or perform initializations that are intended to occur only once when each node starts.
```

File: framework/src/modules/token/commands/transfer_cross_chain.ts [hunk 10]
```diff
@@ -66,6 +81,16 @@ export class TransferCrossChainCommand extends BaseCommand {
 		this._internalMethod = args.internalMethod;
 	}
 
+	/**
+	 * Verifies if:
+	 *  - the token being sent is native to either the sending or the receiving chain
```
Comment (chris529) [inline framework/src/modules/token/commands/transfer_cross_chain.ts:86]:
```suggestion
	 *  - the token being sent is native to either the sending or the receiving chain.
```

File: framework/src/modules/token/commands/transfer_cross_chain.ts [hunk 11]
```diff
@@ -66,6 +81,16 @@ export class TransferCrossChainCommand extends BaseCommand {
...
+	 *  - the right token ID is used to pay the CCM fee
```
Comment (chris529) [inline framework/src/modules/token/commands/transfer_cross_chain.ts:87]:
```suggestion
	 *  - the correct token ID is used to pay the CCM fee.
```

File: framework/src/modules/token/commands/transfer_cross_chain.ts [hunk 12]
```diff
@@ -66,6 +81,16 @@ export class TransferCrossChainCommand extends BaseCommand {
...
+	 *  - and if the sender has enough balance to send the specified amount of tokens.
```
Comment (chris529) [inline framework/src/modules/token/commands/transfer_cross_chain.ts:88]:
```suggestion
	 *  - if the sender has enough balance to send the specified amount of tokens.
```

File: framework/src/modules/token/module.ts [hunk 13]
```diff
@@ -76,6 +76,16 @@ import { CrossChainTransferCommand as CrossChainTransferMessageCommand } from '.
 import { TransferCrossChainCommand } from './commands/transfer_cross_chain';
 import { InternalMethod } from './internal_method';
 
+/**
+ * The Token module handles all logic related to balance.
+ * Specifically the following criteria:
+ *
+ * - Storing token balances of accounts
```
Comment (chris529) [inline framework/src/modules/token/module.ts:83]:
```suggestion
 * - Storing token balances of accounts.
```

**Review by chris529 — COMMENTED** [review]
Looks great! I just made some suggestions 👍

File: framework/src/modules/interoperability/base_cc_command.ts [hunk 14]
```diff
@@ -16,16 +16,42 @@ import { Schema, emptySchema } from '@liskhq/lisk-codec';
 import { NamedRegistry } from '../named_registry';
 import { CCCommandExecuteContext, ImmutableCrossChainMessageContext } from './types';
 
+/**
```
Comment (has5aan) [inline framework/src/modules/interoperability/base_cc_command.ts:19]: The generated documentation for `BaseCCCommand`'s execute and verify methods is somehow not able to link to the `CCCommandExecuteContext<T>` type.

File: framework/src/modules/base_command.ts [hunk 15]
```diff
@@ -17,6 +17,9 @@ import { Schema, emptySchema } from '@liskhq/lisk-codec';
 import { CommandVerifyContext, CommandExecuteContext, VerificationResult } from '../state_machine';
 import { NamedRegistry } from './named_registry';
 
+/**
+ * The `BaseCommand` is the class every module command extends from.
```
Comment (has5aan) [inline framework/src/modules/base_command.ts:21]: A minor point; as `BaseCommand` enforces implementation of an interface, could some info about it be added/rephrased, something of sort; `The `BaseCommand` represents Lisk commands, that each module command extends from.`

File: framework/src/modules/token/commands/transfer_cross_chain.ts [hunk 16]
```diff
@@ -47,13 +47,28 @@ interface Params {
...
 	public schema = crossChainTransferParamsSchema;
 	private _moduleName!: string;
 	private _method!: TokenMethod;
 	private _interoperabilityMethod!: InteroperabilityMethod;
 	private _internalMethod!: InternalMethod;
 
+	/**
+	 * The `init()` hook of a command is called by the Lisk Framework when the node starts.
+	 *
+	 * In this context, you have the opportunity to validate and cache the module config or perform initializations that are intended to occur only once when each node starts.
```
Comment (has5aan) [inline framework/src/modules/token/commands/transfer_cross_chain.ts:66]:
```suggestion
	 * In this context, you have the opportunity to validate and cache the module config or perform initializations that are intended to occur only once.
```

File: framework/src/modules/token/schemas.ts [hunk 17]
```diff
@@ -109,28 +131,44 @@ export const crossChainTransferParamsSchema = {
 		'messageFee',
 		'messageFeeTokenID',
 	],
+	/** A list describing the available parameters for the command. */
 	properties: {
+		/**
+		 * ID of the tokens being transferred.
+		 * `minLength` and `maxLength` are {@link TOKEN_ID_LENGTH}
```
Comment (has5aan) [inline framework/src/modules/token/schemas.ts:138]:
```suggestion
		 * `minLength` and `maxLength` are {@link TOKEN_ID_LENGTH}.
```

File: framework/src/modules/token/schemas.ts [hunk 18]
```diff
@@ -158,31 +196,49 @@ export interface CCTransferMessageParams {
...
+	/** The unique identifier of the schema. */
 	$id: '/lisk/ccTransferMessageParams',
 	type: 'object',
+	/** The required parameters for the command. */
 	required: ['tokenID', 'amount', 'senderAddress', 'recipientAddress', 'data'],
+	/** A list describing the available parameters for the CCM. */
 	properties: {
+		/**
+		 * ID of the tokens being transferred.
+		 * `minLength` and `maxLength` are {@link TOKEN_ID_LENGTH}
```
Comment (has5aan) [inline framework/src/modules/token/schemas.ts:212]:
```suggestion
		 * `minLength` and `maxLength` are {@link TOKEN_ID_LENGTH}.
```

File: framework/src/modules/token/cc_commands/cc_transfer.ts [hunk 19]
```diff
@@ -25,21 +25,48 @@ import { EscrowStore } from '../stores/escrow';
...
+	 * Returns the command name: {@link CROSS_CHAIN_COMMAND_NAME_TRANSFER}
+	 */
 	public get name(): string {
 		return CROSS_CHAIN_COMMAND_NAME_TRANSFER;
 	}
 
+	/**
+	 * The `init()` hook of a command is called by the Lisk Framework when the node starts.
+	 *
+	 * In this context, you have the opportunity to validate and cache the module config or perform initializations that are intended to occur only once when each node starts.
```
Comment (has5aan) [inline framework/src/modules/token/cc_commands/cc_transfer.ts:50]:
```suggestion
	 * In this context, you have the opportunity to validate and cache the module config or perform initializations that are intended to occur only once.
```

**Comment by has5aan** [comment]
Not really sure if this is causing the lint job to fail; but running `yarn format` or `npm run format`, changes are made to `src/modules/token/commands/transfer.ts` that might help fix the lint job.

**Comment by has5aan** [comment]
Perhaps the `docs/` directory entry can also be added to `.gitignore` under `# Distribution directory` section to exclude locally generated docs from git changes.

File: framework/src/modules/base_command.ts (hunk 15 above)
Comment (Tschakki) [reply framework/src/modules/base_command.ts:21]:
```suggestion
 * The `BaseCommand` represents Lisk commands by providing a generic interface, from which each command extends from.
```

File: framework/src/modules/interoperability/base_cc_command.ts (hunk 14 above)
Comment (Tschakki) [reply framework/src/modules/interoperability/base_cc_command.ts:19]: exported it now

**Comment by Tschakki** [comment]
running `yarn format` did not throw any error for me, maybe it is fixed now?

**Comment by has5aan** [comment]
Besides the export of `ImmutableCrossChainMessageContext` all good! 🚀

**Pull request merged**